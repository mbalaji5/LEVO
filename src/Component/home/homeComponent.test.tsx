import React from "react";
import { render, screen } from "@testing-library/react";
import { Home } from ".";

const props = { articles: [] };
const props1 = {
  articles: [
    {
      publishedAt: "2022-01-15T17:02:41.000Z",
      summary: "summary",
      title: "title",
      id: 0,
      url: "",
      imageUrl: "",
      newsSite: "",
      updatedAt: "",
      featured: false,
      launches: [],
      events: []
    }
  ]
};
describe("Should render home Page", () => {
  it("Should validate snapshot", () => {
    const { baseElement } = render(<Home {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
  it("Should validate element", () => {
    render(<Home {...props} />);
    const summaryElement = screen.getByText(/Welcome to LEVO article/i);
    expect(summaryElement).toBeInTheDocument();
    const buttonElement = screen.getByText(/Load More/i);
    expect(buttonElement).toBeInTheDocument();
  });
  it("Should validate element", () => {
    render(<Home {...props1} />);
    const cardItem = screen.getAllByTestId("cardItem");
    expect(cardItem).toBeDefined();
  });
});
