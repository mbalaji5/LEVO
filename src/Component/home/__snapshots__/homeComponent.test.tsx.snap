// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render home Page Should validate snapshot 1`] = `
<body>
  <div>
    <section
      class="home"
    >
      <h2
        class="margin"
      >
        Welcome to LEVO article
      </h2>
      <p
        class="margin"
      >
        Virgin Orbit’s LauncherOne and its 747 carrier aircraft “Cosmic Girl” have launched the US Defense Department’s STP-27VPB mission along with three commercial satellites. The launch occurred on Thursday, January 13 at approximately 2:51 PM PST (22:51 UTC). Cosmic Girl took off from the Mojave Air and Space Port before flying out to the launch zone near the Channel Islands off the southern California coast.
      </p>
      <section
        class="cardContainer"
      />
      <button
        class="loadMore margin"
        disabled=""
      >
        Load More
      </button>
    </section>
  </div>
</body>
`;
